Main/src/ad/j2/hw1/teams/TeamMember.java

```diff
@@ -0,0 +1,44 @@
+package ad.j2.hw1.teams;
+
+public abstract class TeamMember implements Competitor {
+    String type;
+    String name;
+    int maxRunDistance;
+    int maxJumpHeight;
+    String onDistance;
+
+    public boolean isOnDistance(){
+        return onDistance == "Well Done!";
+    }
+
+    public TeamMember(String type, String name, int maxRunDistance, int maxJumpHeight) {
+        this.type = type;
+        this.name = name;
+        this.maxRunDistance = maxRunDistance;
+        this.maxJumpHeight = maxJumpHeight;
+        this.onDistance = "Well Done!";
+    }
+
+    public void run(int distance){
+        if (distance <= maxRunDistance){
+            System.out.printf("%s %s ran OK for distance %s meters\n",type,name,distance);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.